src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -15,13 +15,19 @@ namespace Microsoft.DotNet.Tools.Build
 {
     public class BuildCommand
     {
-        public static int Run(string[] args)
+        public static int Run(string[] args) => Run(args, null);
+
+        public static int Run(string[] args, WorkspaceContext workspace)
```

Comment:
You can't replace build/compile anyways, the only extensibility point here is `dotnet-compile-{compilername}` and csc is almost as second class as everything else 😄.